### ISSUE TITLE
CAS 5.0.0 does not take into account cas.locale.paramName property

### ISSUE THREAD

**Issue by yauheni-sidarenka** [apereo/cas#2200]
I am afraid I cannot use my custom request parameter `lang` to switch to another language on login page in CAS 5.0.0. Seems like `CasWebflowContextConfiguration` should have `@Autowired` field instead of creating its own bean definition of `LocaleChangeInterceptor`. Also I have noticed that `org.apereo.cas.support.pac4j.web.flow.ClientAction` has lines that use Spring's default parameter name for `LocaleChangeInterceptor`:
```
...
restoreRequestAttribute(request, session, LocaleChangeInterceptor.DEFAULT_PARAM_NAME);
...
saveRequestParameter(request, session, LocaleChangeInterceptor.DEFAULT_PARAM_NAME);
...
```
May be this fact also should be considered when resolving this issue.

I have included the following in my report:

- CAS server version: 5.0.0, Gradle overlay
- Container/environment type/version: external Tomcat 8.0.30, Oracle JDK 8u60, Windows 7 Pro OS
- Steps to duplicate the problem: 
1. Set property for desired language parameter name as following:
```
cas.locale.paramName=lang
```
2. Build the project
3. Put resulted `cas.war` file into `CATALINA_HOME/webapps` directory and start the container
4. Go to https://<cas_host>:<cas_port>/cas/login -> login page appears, language of the page is English as expected
5. Go to https://<cas_host>:<cas_port>/cas/login?lang=de -> login page appears, language of the page is still English but German was expected 
6. Go to https://<cas_host>:<cas_port>/cas/login?locale=de -> login page appears, language of the page is German but this parameter should not be processed
- When CAS is starting and `org.springframework` logger is set to `info` level, `cas.log` file says:
```
2016-12-13 18:09:22,889 INFO [org.springframework.beans.factory.support.DefaultListableBeanFactory]
- <Overriding bean definition for bean 'localeChangeInterceptor' with a different definition:
replacing [Root bean: class [org.springframework.aop.scope.ScopedProxyFactoryBean]; scope=;
abstract=false; lazyInit=false; autowireMode=0; dependencyCheck=0; autowireCandidate=true;
primary=false; factoryBeanName=null; factoryMethodName=null; initMethodName=null;
destroyMethodName=null; defined in BeanDefinition defined in class path resource
[org/apereo/cas/config/CasWebAppConfiguration.class]] with [Root bean: class [null]; scope=;
abstract=false; lazyInit=false; autowireMode=3; dependencyCheck=0; autowireCandidate=true;
primary=false; factoryBeanName=casWebflowContextConfiguration;
factoryMethodName=localeChangeInterceptor; initMethodName=null; destroyMethodName=(inferred);
defined in class path resource [org/apereo/cas/config/CasWebflowContextConfiguration.class]]>
```
- Configuration/Overlay pushed to https://github.com/cas-projects/cas-issues/
https://github.com/cas-projects/cas-issues/pull/30

**Comment by mmoayyed** [comment]
This should be fixed in 5.0.2 SNAPSHOT

**Comment by yauheni-sidarenka** [comment]
I have just tried with CAS 5.0.2-SNAPSHOT. The issue has disappeared. Thank you, @mmoayyed !
